server/src/main/java/org/opensearch/search/startree/StarTreeQueryContext.java

```diff
@@ -12,68 +12,148 @@
 
-    /**
-     * Star tree field info
-     * This is used to get the star tree data structure
-     */
-    private final CompositeIndexFieldInfo starTree;
+    private final CompositeDataCubeFieldType compositeMappedFieldType;
 
     /**
-     * Map of field name to a value to be queried for that field
-     * This is used to filter the data based on the query
+     * Cache for leaf results
+     * This is used to cache the results for each leaf reader context
+     * to avoid reading the filtered values from the leaf reader context multiple times
      */
-    private final Map<String, Long> queryMap;
-
-    /**
-    * Cache for leaf results
-    * This is used to cache the results for each leaf reader context
-    * to avoid reading the filtered values from the leaf reader context multiple times
-    */
+    // TODO : Change caching to be based on aggregation specific filters.
     private final FixedBitSet[] starTreeValues;
 
-    public StarTreeQueryContext(CompositeIndexFieldInfo starTree, Map<String, Long> queryMap, int numSegmentsCache) {
-        this.starTree = starTree;
-        this.queryMap = queryMap;
-        if (numSegmentsCache > -1) {
-            starTreeValues = new FixedBitSet[numSegmentsCache];
+    private final QueryBuilder baseQueryBuilder;
+    private StarTreeFilter baseStarTreeFilter;
+
+    // TODO : Implement storing and aggregating aggregation specific filters.
+
+    public StarTreeQueryContext(SearchContext context, QueryBuilder baseQueryBuilder) {
+        this.baseQueryBuilder = baseQueryBuilder;
+        // TODO : We need to select the most appropriate one from multiple star tree field types.
+        compositeMappedFieldType = (CompositeDataCubeFieldType) context.mapperService().getCompositeFieldTypes().iterator().next();
+        // need to cache star tree values only for multiple aggregations
+        boolean cacheStarTreeValues = context.aggregations().factories().getFactories().length > 1;
+        int cacheSize = cacheStarTreeValues ? context.indexShard().segments(false).size() : -1;
+        if (cacheSize > -1) {
+            starTreeValues = new FixedBitSet[cacheSize];
+        } else {
+            starTreeValues = null;
+        }
+    }
+
+    public StarTreeQueryContext(CompositeDataCubeFieldType compositeMappedFieldType, QueryBuilder baseQueryBuilder, int cacheSize) {
+        this.compositeMappedFieldType = compositeMappedFieldType;
+        this.baseQueryBuilder = baseQueryBuilder;
+        if (cacheSize > -1) {
+            starTreeValues = new FixedBitSet[cacheSize];
         } else {
             starTreeValues = null;
         }
     }
 
     public CompositeIndexFieldInfo getStarTree() {
-        return starTree;
+        return new CompositeIndexFieldInfo(compositeMappedFieldType.name(), compositeMappedFieldType.getCompositeIndexType());
     }
 
-    public Map<String, Long> getQueryMap() {
-        return queryMap;
+    public FixedBitSet getStarTreeValue(LeafReaderContext ctx) {
```

Comment:
nit : can we make the naming consistent : 

Technically its getStarTreeValues and getStarTreeValuesArr - since starTreeValues is the name of the variable - we can also rename to be something better like getCachedStarTreeValues/Arr as well